lib/grandpa/commits_tracker.go

```diff
@@ -90,6 +98,9 @@ func (ct *commitsTracker) delete(blockHash common.Hash) {
 // does not exist in the tracker
 func (ct *commitsTracker) message(blockHash common.Hash) (
 	message *CommitMessage) {
+	ct.Lock()
```

Comment:
nit this could be a rw mutex with read lock here

Reply:
https://github.com/ChainSafe/gossamer/pull/2923#discussion_r1021518744